aikau/src/main/resources/alfresco/upload/UploadHistory.js

```diff
@@ -0,0 +1,252 @@
+            index: index
+         };
+         var widgetModel = lang.clone(this.widgetsForUploadTargets);
+         var targetNode = domConstruct.create("div", {
+            className: "alfresco-upload-UploadHistory__target"
+         }, this.domNode);
+         this.processObject(["processCurrentItemTokens"], widgetModel);
+         this.processWidgets(widgetModel, targetNode);
+      },
+
+      /**
+       * Splits the supplied string into an array of NodeRefs (using comma as the delimiter) and then iterates 
+       * over that array calling [createUploadTarget]{@link module:alfresco/upload/UploadHistory#createUploadTarget}
+       * for each NodeRef.
+       * 
+       * @instance
+       * @param {string} value The preference value containing the nodeRefs to create upload targets for
+       */
+      createUploadTargets: function alfresco_upload_UploadHistory__createUploadTargets(value) {
+         if (value)
+         {
+            this.uploadDestinationHistory = value.split(",");
+            array.forEach(this.uploadDestinationHistory, lang.hitch(this, this.createUploadTarget));
+         }
+      },
+
+      /**
+       * Calls the [render]{@link module:alfresco/upload/UploadHistory#render} function to request and
+       * render the user upload history and then sets up a subscription for any further uploads that complete
+       * in order that the display can be refreshed.
+       * 
+       * @instance
+       */
+      postCreate: function alfresco_upload_UploadHistory__postCreate() {
+         this.render();
+         this.alfSubscribe(topics.RELOAD_DATA_TOPIC, lang.hitch(this, this.render));
+      },
+
+      /**
+       * Makes a request for the current users upload history from their preferences and then
+       * with a callback to the [createUploadTargets]{@link module:alfresco/upload/UploadHistory#createUploadTargets}
+       * function to render the upload history.
+       * 
+       * @instance
+       */
+      render: function alfresco_upload_UploadHistory__render() {
+         array.forEach(this.currentHistoryTargets, function(target) {
+            target.destroy();
+         });
+         domConstruct.empty(this.domNode);
+         this.alfPublish(topics.GET_PREFERENCE, {
+            preference: this.preferenceName,
+            callback: this.createUploadTargets,
+            callbackScope: this
+         }, true);
+      },
+
+      /**
+       * This is the widget model used to render each upload target. The root widget should be an
+       * [UploadFolder]{@link moduke:alfresco/upload/UploadFolder} or a widget that extends it.
+       * 
+       * @instance
+       * @type {Array}
+       */
+      widgetsForUploadTargets: [
+         {
+            name: "alfresco/upload/UploadFolder",
+            config: {
+               pubSubScope: "NODE_HISTORY_{index}",
+               nodeRef: "{nodeRef}",
+               xhrRequired: true,
+               widgets: [
+                  {
```

Comment:
The model can be changed, but UploadFolder does set it's own class (which is being used in tests)